src/main/java/mops/presentation/ExceptionController.java

```diff
@@ -0,0 +1,67 @@
+package mops.presentation;
+
+import org.springframework.boot.web.servlet.error.ErrorController;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Exception Handling.
+ */
+@Controller
+public class ExceptionController implements HandlerExceptionResolver, ErrorController {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @SuppressWarnings("PMD.LawOfDemeter")
+    public ModelAndView resolveException(HttpServletRequest request,
+                                         HttpServletResponse response,
+                                         Object handler, Exception ex) {
+
+        String referrer = request.getHeader("referer");
+        ModelAndView mav = new ModelAndView("mops_error");
+        mav.getModel().put("error", ex);
+        mav.getModel().put("referrer", referrer);
+        return mav;
```

Comment:
can you just return "mops_error" and use a Model to put the attributes there

Reply:
nope

Reply:
this is an interface and this requires us to use ModelAndView